ddl/ddl_api.go

```diff
@@ -205,9 +208,36 @@ func (d *ddl) ModifySchemaDefaultPlacement(ctx sessionctx.Context, stmt *ast.Alt
 	return errors.Trace(err)
 }
 
-func (d *ddl) ModifySchemaSetTiFlashReplica(ctx sessionctx.Context, stmt *ast.AlterDatabaseStmt, tiflashReplica *ast.TiFlashReplicaSpec) error {
+func getBatchPendingTiFlashCount(ctx sessionctx.Context) uint32 {
+	return uint32(ctx.GetSessionVars().BatchPendingTiFlashCount)
+}
+
+// getPendingTiFlashTableCount counts unavailable TiFlash replica by iterating all tables in infoCache.
+func (d *ddl) getPendingTiFlashTableCount(sctx sessionctx.Context, originVersion *int64, pendingCount *uint32) {
+	is := d.GetInfoSchemaWithInterceptor(sctx)
+	dbInfos := is.AllSchemas()
+	// If there are no schema change since last time(can be weird)
+	if is.SchemaMetaVersion() == *originVersion {
+		return
+	}
+	cnt := uint32(0)
+	for _, dbInfo := range dbInfos {
```

Comment:
Could we only check current database info?

Reply:
Maybe the user will `ALTER DATABASE` when one `ALTER DATABASE` is processing